pepys_import/core/store/data_store.py

```diff
@@ -1028,13 +1038,89 @@ def get_all_datafiles(self):
+                return self._comment_type_name_dict_on_comment_type_id[comment_type_id]
+            comment_type = (
+                self.session.query(self.db_classes.CommentType)
+                .filter(self.db_classes.CommentType.comment_type_id == comment_type_id)
+                .first()
+            )
+
+            if comment_type:
+                self._comment_type_name_dict_on_comment_type_id[
+                    comment_type_id
+                ] = comment_type.name
+                return comment_type.name
+            else:
+                raise Exception(
+                    "No Comment Type found with Comment type id: {}".format(
+                        comment_type_id
+                    )
+                )
+
+    def get_cached_platform_name(self, sensor_id=None, platform_id=None):
+        """
+        Get platform name from cache on either "sensor_id" or "platform_id"
+        If name is not found in the cache, sytem will load from this data store,
+        and add it into cache.
+        """
+        # invalid parameter handling
+        if sensor_id is None and platform_id is None:
+            raise Exception(
+                'either "sensor_id" or "platform_id" has to be provided to get "platform name"'
+            )
+
+        if sensor_id:
+            # return from cache
+            if sensor_id in self._platform_dict_on_sensor_id:
+                return self._platform_dict_on_sensor_id[sensor_id]
+            sensor = (
+                self.session.query(self.db_classes.Sensor)
+                .filter(self.db_classes.Sensor.sensor_id == sensor_id)
+                .first()
+            )
+
+            if sensor:
+                platform_id = sensor.host
+            else:
+                raise Exception("No sensor found with sensor id: {}".format(sensor_id))
+
+        if platform_id:
+            # return from cache
+            if platform_id in self._platform_dict_on_platform_id:
+                return self._platform_dict_on_platform_id[platform_id]
+            platform = (
+                self.session.query(self.db_classes.Sensor)
+                .filter(self.db_classes.Sensor.sensor_id == sensor_id)
+                .first()
+            )
+
+            if platform:
+                self._platform_dict_on_platform_id[platform_id] = platform.name
+                if sensor_id:
+                    self._platform_dict_on_sensor_id[sensor_id] = platform.name
+            else:
+                raise Exception(
+                    "No Platform found with platform id: {}".format(platform_id)
+                )
+        return platform.name
+
+    def export_datafile(self, datafile_id, datafile):
         """
         Get states, contacts and comments based on Datafile ID.
 
         :param datafile_id:  ID of Datafile
         :type datafile_id: String
         """
+        f = open("{}.rep".format(datafile), "w+")
```

Comment:
could you pre-process the `datafile`, to swap `.` for `_` - this will keep it as a valid filename after we've appended `.rep`